src/object/cli_obj.c

```diff
@@ -2442,6 +2442,38 @@ obj_csum_update(struct dc_object *obj, daos_obj_update_t *args,
 	if (!daos_csummer_initialized(csummer)) /** Not configured */
 		return 0;
 
+	if (!daos_csummer_get_csum(csummer) &&
+	    daos_csummer_get_dedup(csummer)) {
+		uint32_t	dedup_th = daos_csummer_get_dedupsize(csummer);
+		int		i;
+		bool		candidate = false;
+
+		/**
+		 * Checksums are only enabled for dedup purpose.
+		 * Verify whether the I/O is a candidate for dedup.
+		 * If not, then no need to provide a checksum to the server
+		 */
+
+		for (i = 0; i < args->nr; i++) {
+			daos_iod_t	*iod = &args->iods[i];
+			int		 j = 0;
+
+			if (iod->iod_type == DAOS_IOD_SINGLE)
+				/** dedup does not support single value yet */
+				return 0;
+
+			for (j = 0; j < iod->iod_nr; j++) {
+				daos_recx_t	*recx = &iod->iod_recxs[j];
+				if (recx->rx_nr * iod->iod_size >= dedup_th)
```

Comment:
(style)  'dedup' may be misspelled - perhaps 'dedupe'?
(style)  Missing a blank line after declarations